docs/UpgradingToJest30.md

```diff
@@ -0,0 +1,206 @@
+- The minimum TypeScript version is now 5.0. Support for TypeScript 4.3 and below has been removed. Update your TypeScript dependency to v5+ if you use TypeScript with Jest.
+- The `jest-environment-jsdom` package now uses JSDOM v22. This upgrade may introduce behavior changes in the DOM environment. If you encounter differences in DOM behavior or new warnings, refer to the JSDOM release notes for [v21–22](https://github.com/jsdom/jsdom/compare/21.0.0...22.0.0).
+
+## Configuration Updates
+
+### Support for `.mts` and `.cts` File Extensions
+
+Jest 30 expands support for ESM and TypeScript module file extensions:
+
+- The default `moduleFileExtensions` now include `.mts` and `.cts` (TypeScript ESM and CommonJS modules) in addition to the usual extensions.
+- The default `testMatch` and `testRegex` patterns have been updated to recognize `.mjs`, `.cjs`, `.mts`, and `.cts` files as test files.
+
+:::info
+
+If your project contains files with these extensions that are **not** intended to be treated as modules or tests, you may need to adjust your configuration. Conversely, if you have test files with these extensions, Jest will now detect them by default (you may remove custom configuration that was previously needed to include them).
+
+:::
+
+### `testPathPattern` Renamed to `testPathPatterns`
+
+If you filter tests by path, note that the configuration option and CLI flag have changed:
+
+- **Configuration:** The `testPathPattern` option (singular) has been replaced by `testPathPatterns` (plural). Instead of a single regex string, this option now takes an array of patterns. For example:
+
+  ```js
+  // Jest 29 configuration
+  testPathPattern: "e2e/.*\\.spec\\.js",
+  
+  // Jest 30 configuration
+  testPathPatterns: ["e2e/.*\\.spec\\.js"]
+  ```
+
+  Each pattern in the array is treated as a regex or glob to match test file paths.
+
+- **CLI usage:** The `--testPathPattern` flag is now `--testPathPatterns`. You can pass multiple patterns by separating them with spaces or by repeating the flag. For example:
+
+  ```bash
+  # Old (Jest 29)
+  jest --testPathPattern="unit/.*" 
+  
+  # New (Jest 30)
+  jest --testPathPatterns "unit/.*" "integration/.*"
+  ```
+
+  Internally, Jest consolidates these patterns into a `TestPathPatterns` object. If you were programmatically calling Jest’s watch mode with a `testPathPattern`, you must now construct a `TestPathPatterns` instance instead.
+
+### Removed `--init` Command
+
+The interactive config initialization command `jest --init` has been **removed**. This deprecated command was used to scaffold a Jest configuration file. If you need to create a config, you should now manually create a `jest.config.js` (or `.ts`/`.mjs`) file or consult the documentation for configuration options. Existing projects are not affected except that you can no longer run `jest --init` in new projects.
```

Comment:
There is no need to create config files manually. The behaviour of `jest --init` was move to `create-jest` package: https://github.com/jestjs/jest/tree/main/packages/create-jest

Reference: https://github.com/jestjs/jest/pull/14453